### DISCUSSION TITLE
Get all tags from an item

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#1558] [Help] [answered]
Hey, I want to get every tag from an item. How can I do that?

**Comment by TheMode** [comment, marked as answer]
You can retrieve the item NBT using `ItemStack#toItemNBT` or from the meta.

There is no way to retrieve a list of tags, as they are only used to mutate NBT.

**Reply by DevBlokJochem** [reply]
Can you help me a little bit with that?
I have tried to convert the itemNBT to a json string, that works. But how can I get the tags out of the string?

**Reply by TheMode** [reply]
The tags are not stored in the NBT. They serve as a key to access a part of the compound.

**Reply by DevBlokJochem** [reply]
Im creating an item editor, and I want to see/edit the tags that are on the items.  
For me, it is enough to get the key and value.
If im printing the compoundentry's of the itemNBT, I got this:
```
Count 1
tag {display={"Name":"{\"text\":\"hallo\"}","Lore":["{\"text\":\"test\"}"]}, test2="test4"}
id minecraft:diamond_axe
``` 
and I want to create a map that has for this example:

```
display
  "name" - "hallo"
  "lore" - "test"
"test2" - "test4
```

because Ive set these tags on the item.

**Reply by TheMode** [reply]
I repeat, the item does NOT store tags, but NBT.

You can iterate through the compound fields and create a map from that. It is completely unrelated to the tag API.
